src/containers/AppCognito.tsx

```diff
@@ -39,35 +38,53 @@ const AppCognito: React.FC = /* istanbul ignore next */ () => {
 
   const [, , bankAccountsService] = useMachine(bankAccountsMachine);
 
-  const { route, signOut, user } = useAuthenticator();
+  const isLoggedIn =
+    authState.matches("authorized") ||
+    authState.matches("refreshing") ||
+    authState.matches("updating");
 
   useEffect(() => {
-    console.log("auth route: ", route);
-    if (route === "authenticated") {
-      authService.send("COGNITO", { user });
+    if (!isLoggedIn) {
+      fetchAuthSession().then((authSession) => {
+        if (authSession && authSession.tokens && authSession.tokens.accessToken) {
+          const { tokens, userSub } = authSession;
+          authService.send("COGNITO", {
+            accessTokenJwtString: tokens!.accessToken.toString(),
+            userSub: userSub!,
+            email: tokens!.idToken!.payload.email,
+          });
+        } else {
+          void signInWithRedirect();
+        }
+      });
     }
-  }, [route, user]);
+  }, [isLoggedIn]);
 
   useEffect(() => {
     authService.onEvent(async (event) => {
-      if (event.type === "done.invoke.performLogout") {
-        console.log("AppCognito authService.onEvent done.invoke.performLogout");
+      if (
+        event.type === "done.invoke.performLogout" ||
+        // we want the client-side app to discard its JWTs even if server-side errors out:
+        event.type.startsWith("error.platform.authentication.logout")
+      ) {
+        console.log(
+          "AppCognito authService.onEvent done.invoke.performLogout|error.platform.authentication.logout"
+        );
         await signOut();
       }
     });
-  }, [signOut]);
+  }, []);
 
-  const isLoggedIn =
-    authState.matches("authorized") ||
-    authState.matches("refreshing") ||
-    authState.matches("updating");
+  if (!isLoggedIn) {
+    return null;
+  }
 
-  return isLoggedIn ? (
+  return (
     <div className={classes.root}>
       <CssBaseline />
 
       <PrivateRoutesContainer
-        isLoggedIn={isLoggedIn}
+        isLoggedIn={true}
```

Comment:
is there a reason we just pass `true` into this or is this left over from development?

Reply:
Ahh, this (I think) is a result of the change in the purpose of version 5 of the amplify-react-ui Authenticator component from what was done in version 1 of that library (which did not use an Authenticator component) and from what I suppose (though haven't looked closely) is done with okta, google, and auth0 in their tests presently.  The pre-vite (create-react-app) incarnation using version 1 of amplify-react-ui I believe intended to provide a login prompt for use _without redirection_, and the redirection would only occur once "Log in with AWS" was pressed within that prompt. The login prompt wasn't tested but if it were, would've been a direct login using ordinary authorization as coded in the express backend.

There was no similar login prompt provided by version 5 of amplify-react-ui, other than the Authenticator component, which is not designed to also provide a link to redirect to the hosted UI.  So, since it wasn't tested anyway, I figured `AppCognito.tsx` would no longer offer that direct-login UI and instead redirect immediately to the hosted UI (line 57) upon page load, if unauthenticated.  The rendering of `AppCognito.tsx` is foregone until the redirect occurs (line 79).  As such, if `PrivateRoutesContainer` here is rendered at all, it is because `null` was not returned on line 79, because `isLoggedIn` is already true (post-redirect-and-redirect-back from the hosted UI).  `isLoggedIn` could still be passed here, but it will always be `true`.

Reply:
probably just makes sense to just pass `true` since we do share this component with the other auth tests. That seems fine